### PR TITLE
Removed access to $strPickerTable instead of DcaFilterInterface all over DC files

### PR DESCRIPTION
**Pull request by aschempp**

### CODE REVIEW

File: src/Resources/contao/drivers/DC_Folder.php [hunk 1]
```diff
@@ -2751,7 +2757,7 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 
 				if ($this->strPickerField)
 				{
-					$_buttons .= $this->getPickerInputField($currentEncoded, ($GLOBALS['TL_DCA'][$this->strPickerTable]['fields'][$this->strPickerField]['eval']['files'] || $GLOBALS['TL_DCA'][$this->strPickerTable]['fields'][$this->strPickerField]['eval']['filesOnly']) ? '' : ' disabled');
+					$_buttons .= $this->getPickerInputField($currentEncoded);
```
Comment (leofeyer) [inline src/Resources/contao/drivers/DC_Folder.php]: Why did you remove the `$this->blnFilesOnly ? '' : 'disabled'` part?

File: src/Resources/contao/drivers/DC_Folder.php [hunk 2]
```diff
@@ -2571,7 +2577,7 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 				{
 					--$countFiles;
 				}
-				elseif ($this->blnHideFiles && !is_dir(TL_ROOT . '/' . $currentFolder . '/' . $file))
+				elseif (!$this->blnShowFiles && !is_dir(TL_ROOT . '/' . $currentFolder . '/' . $file))
```
Comment (leofeyer) [inline src/Resources/contao/drivers/DC_Folder.php]: Why did you change the "hide files" logic?

File: src/Resources/contao/drivers/DC_Folder.php (hunk 1 above)
Comment (aschempp) [inline src/Resources/contao/drivers/DC_Folder.php]: Because that line is only executed if there are files, and if there are files they are always selectable. `blnFilesOnly` only defines that folders are not to be selected.

File: src/Resources/contao/drivers/DC_Folder.php (hunk 2 above)
Comment (aschempp) [inline src/Resources/contao/drivers/DC_Folder.php]: That's how it was always implemented. It's `eval => ['files' => true]`, and by default it's `false`.

File: src/Resources/contao/drivers/DC_Folder.php [hunk 3]
```diff
@@ -2661,7 +2667,7 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 			}
 		}
 
-		if ($this->blnHideFiles)
+		if (!$this->blnFiles)
```
Comment (leofeyer) [inline src/Resources/contao/drivers/DC_Folder.php]: This is wrong and should be `if (!$this->blnFiles && ! $this->blnFilesOnly)`. See:

https://github.com/contao/core-bundle/blob/master/src/Resources/contao/widgets/FileSelector.php#L535
